src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+                                                (1 - 2 * minorAlleleFraction));
+                        return num / denom;
+                    } else { /* breakpointHetAlleleFraction < minorAlleleFraction < 1/2 */
+                        double denom = 2 * FastMath.pow((1 - minorAlleleFraction) * minorAlleleFraction * maxCopyNumber, 2) *
+                                maxAbnormalFraction * minAbnormalFraction * (maxAbnormalFraction - minAbnormalFraction);
+                        double num = (maxAbnormalFraction - minAbnormalFraction) * (-1 + 2 * minorAlleleFraction +
+                                (1 + minorAlleleFraction * (-2 + maxCopyNumber)) * (-1 + minorAlleleFraction * maxCopyNumber) *
+                                maxAbnormalFraction * minAbnormalFraction) + 2 * (1 + minorAlleleFraction * (-2 + minorAlleleFraction *
+                                maxCopyNumber)) * maxAbnormalFraction * minAbnormalFraction * FastMath.log(maxAbnormalFraction / minAbnormalFraction);
+                        return num / denom;
+                    }
+                })
+                .boxed().collect(Collectors.toList()));
+
+        /* calculate the log prior */
+        alleleFractionLogPriors.clear();
+        alleleFractionLogPriors.addAll(alleleFractionPriors.stream()
+                .mapToDouble(FastMath::log).boxed().collect(Collectors.toList()));
+    }
+
+    /**
+     * calculate the log probability, safeguarded with a minimum probability DEFAULT_MIN_BASE_ERROR_THIRD
+     * @param prob probability
+     * @return safeguarded-log of probability
+     */
+    private double getSafeguardedLogProbability(final double prob) {
+        return FastMath.log(FastMath.max(DEFAULT_MIN_BASE_ERROR_THIRD, prob));
+    }
+
+    /**
+     * calculate the log likelihood of a SNP site being homozygous for a given read pileup
+     * (see CNV-method.pdf for details)
+     * @param baseQualities map of bases to list of their calling error probabilities at the SNP site
+     * @param alleleRef the ref allele base
+     * @param alleleAlt the alt allele base
+     * @param homRefPrior the prior probability of the ref allele given that the site is homozygous
+     * @return the log likelihood
+     */
+    @VisibleForTesting
+    public double getHomLogLikelihood(final Map<Nucleotide, ArrayList<BaseQuality>> baseQualities,
+                                      final Nucleotide alleleRef, final Nucleotide alleleAlt,
+                                      final double homRefPrior) {
+
+        /* initilize the log likehoods of ref and alt with the priors ... */
+        double homRefLogLikelihood = FastMath.log(Utils.nonNull(homRefPrior));
+        double homAltLogLikelihood = FastMath.log(1.0 - Utils.nonNull(homRefPrior));
+
+        /* ... and add the log likelihood of the reads */
+        for (Nucleotide base : baseQualities.keySet()) {
```

Comment:
I don't think it is necessary because in principle, `getHomLogLikelihood' and `getHetLogLikelihood' are private and baseQualities are internally passed such that they can never be `null'.